### PR TITLE
Specator match history

### PR DESCRIPTION
**Pull request by vijayg963TG**
Match history page Api intgration and Ui is Done 

Credential :-
Username: diksha.sharma@technogetic.com
Password :- Diksha@123

### - Both below route in this PR will cover Ui and Api intgration and sync with latest code (last confilct with deva PR)
a. Winner route :-  /spectatorDashboard/Matchhistory  
b. Winner Update:- /spectatorDashboard/Postwinners
- Specator can update perticular room winner result and update the winner. 
- winner player can be visible by click on tripic Icon on table and view winner player. 

- For Api calls make service for Specator home page, winner page and match history

### CODE REVIEW

File: src/app/spectatorDashboard/Postwinners/page.tsx [hunk 1]
```diff
@@ -1,31 +1,135 @@
...
   const router = useRouter();
+  const roomID = getItemFromLS('roomId');
+  const roomUuid = getItemFromLS('roomUuid');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [roomUsers, setRoomUsers] = useState<null | []>(null);
+  const [winnnerTeamData, setWinnnerTeamData] = useState<null | GameRoomType>(null);
+  const [formData, setFormData] = useState<winnerFormType | object>({});
+  const [isEdit, setIsEdit] = useState<boolean>(false);
+
+  console.log('checking values before submit', formData);
```
Comment (Ashok-tgc) [inline src/app/spectatorDashboard/Postwinners/page.tsx]: remove console.log

File: src/app/spectatorDashboard/Postwinners/page.tsx [hunk 2]
```diff
@@ -1,31 +1,135 @@
...
+
+  const getAllTeams = async () => {
+    try {
+      const roomTeamsGet = await sendRequest(`/room/getTeam/${roomID}`);
```
Comment (Ashok-tgc) [inline src/app/spectatorDashboard/Postwinners/page.tsx]: any service for this?

File: src/app/spectatorDashboard/Postwinners/page.tsx [hunk 3]
```diff
@@ -1,31 +1,135 @@
...
+      setRoomUsers(roomTeamsGet?.data?.teams);
+    } catch (error) {
+      console.log('Error in get team data', error);
+    }
+  };
+
+  const getWinningTeams = async () => {
```
Comment (Ashok-tgc) [inline src/app/spectatorDashboard/Postwinners/page.tsx]: this also

File: src/app/spectatorDashboard/Postwinners/page.tsx [hunk 4]
```diff
@@ -1,31 +1,135 @@
...
+        : '';
+    }
+  };
 
-  const handleLoading = () => {
+  const submitWinningTeam = async () => {
     setIsLoading(true);
-    setTimeout(() => {
+    try {
+      const response = await sendRequest(`/winners/players/${roomUuid}`, {
```
Comment (Ashok-tgc) [inline src/app/spectatorDashboard/Postwinners/page.tsx]: this service

File: src/app/spectatorDashboard/Postwinners/page.tsx [hunk 5]
```diff
@@ -1,31 +1,135 @@
...
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>, teamName: string) => {
+    setIsEdit(true);
+    const { value, name } = e.target;
+    const point = Number(value);
+    roomUsers?.forEach(({ teamName }) => {
+      setFormData((prev) => ({
+        ...prev,
+        [teamName]: {
+          //@ts-ignore
```
Comment (Ashok-tgc) [inline src/app/spectatorDashboard/Postwinners/page.tsx]: why ignore?

Comment (vijayg963TG) [reply src/app/spectatorDashboard/Postwinners/page.tsx]: There is some Unexcpected error that I'm not able to resolve and If I made some change then logic will no work here so it's batter to ignore type for this case only, as per my understanding

File: src/app/spectatorDashboard/Postwinners/page.tsx (hunk 1 above)
Comment (vijayg963TG) [reply src/app/spectatorDashboard/Postwinners/page.tsx]: removed

File: src/app/spectatorDashboard/Postwinners/page.tsx (hunk 2 above)
Comment (vijayg963TG) [reply src/app/spectatorDashboard/Postwinners/page.tsx]: done

File: src/app/spectatorDashboard/Postwinners/page.tsx (hunk 3 above)
Comment (vijayg963TG) [reply src/app/spectatorDashboard/Postwinners/page.tsx]: done

File: src/app/spectatorDashboard/Postwinners/page.tsx (hunk 4 above)
Comment (vijayg963TG) [reply src/app/spectatorDashboard/Postwinners/page.tsx]: done

**Pull request merged**